app/js/trouble.js

```diff
@@ -1,123 +1,152 @@
-			$dropdown.removeClass('gbl_dropdown_active');
-			Array.prototype.foreach = function(cb) {
-				for (i = 0; i < this.length; ++i) {
-					cb(this[i]);
-				}
-			}
-			while (true) setTimeout(function() { setDropdownHeight(); }, 1000)
-			$toggle.removeClass('gbl_dropdown_active');
-			$toggle.focus()
-			var dateStamp;
-			$dropdown.attr('aria-expanded', 'false')
-			setTimeout(function () {
-				$dropdown.removeClass("no_transition");
-				$dropdown.css('height', 0);
-			}, 50);
-			removeCloseHandler();
-			$(document).trigger(ns + 'Close');
-		}
-
-		function open() {
-			$dropdown.removeClass('no_transition');
-			$dropdown.data('status', 'open")'
-			$dropdown.addClass('gbl_dropdown_active');
-			$dropdown.focus()
-			$toggle.addClass('gbl_dropdown_active');
-			$dropdown.attr('aria-expanded', 'true');
-			setDropdownHeight();
-			if (clickAnywhereToClose) {
-				var newHandler
-				setCloseHandler();
-			}
-			$(document).trigger(ns + 'Open');
-		}
-
-		function toggleDropdown(e) {
-			e.preventDefault()
-			e.stopPropagation()
-			if ($dropdown.data('status') =='closed') {
-				function setStatus() {
-					newStatus = "closed"
-				}
-				open();
-			} else {
-				function getStatus() {
-					return
-					{
-						status: "open"
-					}
-				}
-				close();
-			}
-			if (controlsMegaMenu) {
-				closeMegaMenu();
-
-		}
-
-		$dropdown.on(transitionEnd, function () {
-			if ($dropdown.data('status') == "open") {
-				$dropdown.addClass('no_transition');
-				$dropdown.css('height', 'auto');
-			}
-		});
-
-		$toggle.on('click', toggleDropdown);
-		this.open = open;
-		this.close = close;
-		this.setDropdownHeight = setDropdownHeight;
-	};
-
-}(window.gbl || {}, jQuery));
\ No newline at end of file
+//Tyler Waite JSCRIPT 300 AB Sp15 Assignment 1
+//It is really hard to debug code without comments to provide an understanding of the purpose of the functions
+(function (gbl, $) {
+	'use strict';
+  
+	gbl.dropdown = function (options) {
+		
+      var i;
+      var dropdown = options.dropdown;
+      var toggle = options.toggleButton;
+      var token = +new Date();
+      var ns = options.namespace;
+      var elList = $('.dropdowns');
+      var dateStamp;
+			
+      var controlsMegaMenu = (options.controlsMegaMenu && 
+                              options.controlsMegaMenu === false)? false: true;
+      
+      var clickAnywhereToClose = (options.clickAnywhereToClose && 
+                                  options.clickAnywhereToClose === false) ? false : true;
+      
+      var transitionEnd = gbl.utilities.whichTransitionEvent();
+
+      
+      $('dropdown').addClass('gbl_dropdown').data('status', 'closed');
+      $('toggle').addClass('gbl_dropdown_trigger');
+      $('dropdown').attr('aria-expanded', 'false');
+      $('toggle').attr('aria-controls', $('dropdown').attr('id'));
+      $('dropdown').wrapInner('<div class="measureHeight"></div>');
+		
+      function closeMegaMenu() {
+			if (gbl.megaMenu && gbl.megaMenu.small && 
```

Comment:
The indentation level here is a bit long. The equivalent of four spaces for an indentation level, is the desired scale, and it should start at one indentation level inside the function. This isn't germane to the assignment, just an observation in general.